client/core/core.go

```diff
@@ -2309,6 +2345,76 @@ func handleRevokeMatchMsg(_ *Core, dc *dexConnection, msg *msgjson.Message) erro
 	return tracker.db.UpdateOrder(metaOrder)
 }
 
+// handleTradeSuspensionMsg is called when a trade suspension notification is received.
+func handleTradeSuspensionMsg(c *Core, dc *dexConnection, msg *msgjson.Message) error {
+	var sp msgjson.TradeSuspension
+	err := msg.Unmarshal(&sp)
+	if err != nil {
+		return fmt.Errorf("trade suspension unmarshal error: %v", err)
+	}
+
+	dc.pendingSuspendsMtx.Lock()
+	defer dc.pendingSuspendsMtx.Unlock()
+
+	// Attempt to cancel and remove pending suspends.
+	if sched := dc.pendingSuspends[sp.MarketID]; sched != nil {
+		if !sched.Stop() {
+			return fmt.Errorf("market %s for dex %s is already suspended",
+				sp.MarketID, dc.acct.url)
+		}
+		delete(dc.pendingSuspends, sp.MarketID)
+	}
+
+	// Set the new scheduled suspend.
+	duration := time.Until(encode.UnixTimeMilli(int64(sp.SuspendTime)))
+	dc.pendingSuspends[sp.MarketID] = time.AfterFunc(duration, func() {
+		dc.pendingSuspendsMtx.Lock()
+		dc.suspend(sp.MarketID)
+		if bookie := dc.books[sp.MarketID]; bookie != nil {
+			bookie.Reset()
+		}
+		if !sp.Persist {
+			assets := strings.Split(sp.MarketID, "_")
+			base, ok := dex.BipSymbolID(assets[0])
+			if !ok {
+				log.Errorf("no ID for base BIP symbol: %v", assets[0])
+			}
+
+			quote, ok := dex.BipSymbolID(assets[1])
+			if !ok {
+				log.Errorf("no ID for quote BIP symbol: %v", assets[1])
+			}
```

Comment:
Shouldn't parse the base and quote IDs from the `MarketID`. Instead, grab them from the market in the `marketMap`. This way would work for now, but the spec does not guarantee that the `MarketID` will be constructed from the ticker symbols.